wallettool/src/main/java/org/bitcoinj/wallettool/WalletTool.java

```diff
@@ -174,6 +175,8 @@ public class WalletTool implements Callable<Integer> {
     private String actionStr;
     @CommandLine.Option(names = "--net", description = "Which network to connect to. Valid values: ${COMPLETION-CANDIDATES}. Default: ${DEFAULT-VALUE}")
     private Network net = Network.MAIN;
+    @CommandLine.Option(names = "--structure", description = "Which HD hierarchy structure to use. Valid values: ${COMPLETION-CANDIDATES}. Default: ${DEFAULT-VALUE}")
+    private Structure structure = Structure.BIP43;
```

Comment:
Since this is the default, I think it should be in sync with `KeyChainGroupStructure.DEFAULT`.

Reply:
I thought I had force-pushed a revision to make this `BIP32`. I'll check and see what happened.

I think we agreed elsewhere that `wallet-tool` should for now default to `BIP32` and I'll find that commit and force-push it. But, in general, I think **bitcoinj** applications should not use `DEFAULT` and potentially have their behavior changed by an upgrade. In fact I think that `DEFAULT` should be deprecated in either this release or the next one.

Reply:
I force-pushed that the default for this app is now `BIP32`.

Do you agree that this app should not use `KeyChainGroupStructure.DEFAULT` at all? Should we make a separate PR to deprecated `KeyChainGroupStructure.DEFAULT`?

Reply:
> Do you agree that this app should not use KeyChainGroupStructure.DEFAULT at all?

Yes, I think that's reasonable.

> Should we make a separate PR to deprecated KeyChainGroupStructure.DEFAULT?

We could, but we also have constructors with defaults for the structure. Are they all deprecated as well?

> applications should not use DEFAULT and potentially have their behavior changed by an upgrade

That's the reason I'd be, to say the least, very reluctant to change the default.

Reply:
> > Should we make a separate PR to deprecated KeyChainGroupStructure.DEFAULT?
> 
> We could, but we also have constructors with defaults for the structure. Are they all deprecated as well?

I've thought about this more and think we should definitely deprecate it and submitted PR #2443 to do so. Take a look at the PR, I think you'll agree it's the right way to go. The existing constructors, static constructors, and builders that use `DEFAULT` have been changed to use `BIP32` and I think that makes the code more readable. I don't think we need to deprecate any of the existing constructors, static constructors, and builders that use BIP32 (formerly DEFAULT) in this release, but we might choose to do so at some point in the future.

> 
> > applications should not use DEFAULT and potentially have their behavior changed by an upgrade
> 
> That's the reason I'd be, to say the least, very reluctant to change the default.

Agree, we should not change the behavior of `KeyChainGroupStructure.DEFAULT`. We should deprecate it, but we should never change behavior.

Reply:
I think this conversation can be resolved?